lib/portlayer/exec/container.go

```diff
@@ -240,6 +241,37 @@ func (c *Container) stop(ctx context.Context) error {
 	return c.vm.WaitForPowerState(ctx, types.VirtualMachinePowerStatePoweredOff)
 }
 
+func (c *Container) startGuestProgram(ctx context.Context, name string, args string) error {
+	o := guest.NewOperationsManager(c.vm.Client.Client, c.vm.Reference())
+	m, err := o.ProcessManager(ctx)
+	if err != nil {
+		return err
+	}
+
+	spec := types.GuestProgramSpec{
+		ProgramPath: name,
+		Arguments:   args,
+	}
+
+	auth := types.NamePasswordAuthentication{
+		Username: c.ExecConfig.ID,
+	}
+
+	_, err = m.StartProgram(ctx, &auth, &spec)
+
+	return err
+}
+
+func (c *Container) Signal(ctx context.Context, num int64) error {
+	defer trace.End(trace.Begin("Container.Signal"))
+
+	if c.vm == nil {
+		return fmt.Errorf("vm not set")
+	}
+
+	return c.startGuestProgram(ctx, "kill", fmt.Sprintf("%d", num))
```

Comment:
Huh, so this is one of the toolbox's program and not linux's kill, right?   The reason I ask is you're calling kill with the args == signal value.